js/angular/services/foundation.core.animation.js

```diff
@@ -56,16 +56,21 @@
       element[0].style.transitionDuration = '';
       element.addClass(activeClass);
 
-      element.one(events.join(' '), function() {
-        finishAnimation();
-      });
+      element.on(events.join(' '), eventHandler);
 
-      setTimeout(function() {
+      var animationTimeout = setTimeout(function() {
         if(timedOut) {
           finishAnimation();
         }
       }, 3000);
 
+      function eventHandler(e) {
+        if (element[0].id === e.target.id) {
```

Comment:
Hey thanks for the +1, `this` may equal to the animation service, would reverting back to `element[0] === e.target` be enough to follow the jQuery doc?